### PR TITLE
Adding a PumpManager class

### PR DESCRIPTION
**Pull request by mtlynch**
Adds a PumpManager class with unit tests. One unit test works, two fail, and
one is not implemented yet.

### CODE REVIEW

File: greenpithumb/pump_manager.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        """Run the water pump if there is a need to run it.
+
+        Checks whether it's time to run the pump and runs it if needed.
+        Running the pump is necessary if the soil's moisture is below the
+        defined moisture threshold.
+
+        """
+        # TODO(JeetShetty): Add logic here so that it only runs the pump when
+        # the soil moisture is too low.
+        self._pump.pump_water(DEFAULT_PUMP_AMOUNT)
```
Comment (JeetShetty) [inline greenpithumb/pump_manager.py]: How can we do this without having defined a pump_water method?

File: tests/test_pump_manager.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+from __future__ import absolute_import
+import unittest
+
+import mock
+
+from greenpithumb import pump_manager
+
+
+class PumpManagerTest(unittest.TestCase):
+
```
Comment (JeetShetty) [inline tests/test_pump_manager.py]: This is pretty foreign/confusing to me and I'm having a tough time debugging and figuring out why my change is failing the "moisture_is_low" case as a result of that.

File: greenpithumb/pump_manager.py (hunk 1 above)
Comment (JeetShetty) [inline greenpithumb/pump_manager.py]: if self._moisture_sensor < SOIL_MOISTURE_THRESHOLD:
            self._pump.pump_water(DEFAULT_PUMP_AMOUNT)

Comment (mtlynch) [inline greenpithumb/pump_manager.py]: It's missing a method. `self._moisture_sensor` is an object, so we need to call its `.moisture()` method to get the moisture, but once you do that it should work.

Also if you surround your code with triple backticks (\`\`\`), it formats nicely on Github comments

**Pull request merged**